src/apps/api/views/leaderboards.py

```diff
@@ -43,15 +44,24 @@ def update(self, request, *args, **kwargs):
         return response
 
 
+def is_user_submission_owner_or_admin(request, submission, competition):
+    if request.user.is_superuser or request.user in competition.collaborators.all():
+        return True
+    elif request.user == submission.owner:
+        return True
+    else:
+        raise drf_exceptions.AuthenticationFailed("You do not have permission to change this submission's leaderboard status")
```

Comment:
It seems superior to `404` when the requestor is not the owner rather than `403`, because then we hide the fact that the submission even exists. Let's check if the requestor is the owner inside of our view rather than the permission class.

Reply:
Changed the error to 404